### PR TITLE
Add Kerr plus scalar wave initial data for ScalarTensor

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

Create a new folder in AnalyticData for the ScalarTensor system.

Since we don't yet solve the constraint equations numerically, we prescribe analytic initial data that does not take into account the backreaction of the scalar. Here we add data for Kerr with a spherical harmonic scalar wave.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/GhScalarTensor/Factory.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace gh::AnalyticData {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GhScalarTensor/Factory.hpp]: I would flip the namespaces, `gh::ScalarTensor::AnalyticData`. Probably need:
```cpp
/// ScalarTensor solutions wrapped for GH
namespace gh::ScalarTensor::AnalyticData {
``` 
I'm not sure why GRMHD is done this way, that needs to get fixed but not your fault :) (looks like it's my fault, actually :( sigh...)

File: src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma:  no_include "DataStructures/Tensor/TypeAliases.hpp"
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp]: [optional] We stopped using IWYU a while back, so you can just exclude these if you want. up to you :)

File: src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+namespace ScalarTensor::AnalyticData {
+
+KerrSphericalHarmonic::KerrSphericalHarmonic(
+    double mass, const std::array<double, 3>& dimensionless_spin,
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp]: `const double` x 4 & 1 `const std::pair`

File: src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+template <typename DataType>
+tuples::TaggedTuple<CurvedScalarWave::Tags::Pi>
+KerrSphericalHarmonic::variables(
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<CurvedScalarWave::Tags::Pi> /*meta*/) const {
+  Scalar<DataType> pi =
+      make_with_value<Scalar<DataType>>(x, 0.0);
+  get(pi) += get(magnitude(x)) - radius_;
+  get(pi) = exp(-get(pi) * get(pi) / width_sq_);
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp]: [optional] `get(pi) * get(pi)` -> `square(get(pi))`

File: src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  const Spectral::Swsh::SpinWeightedSphericalHarmonic spherical_harmonic(
+      0, mode_.first, mode_.second);
+  const auto theta = atan2(hypot(x[0], x[1]), x[2]);
+  const auto phi = atan2(x[1], x[0]);
+  get(pi) *= real(spherical_harmonic.evaluate(theta, phi, sin(theta / 2.),
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.cpp]: `theta / 2.` -> `0.5 * theta` Division is ~ 25x slower than multiplication, and doing scalar math to the farthest left allows Blaze to do some optimizations (not always and I think not here, but just a good habit)

File: tests/Unit/PointwiseFunctions/AnalyticData/ScalarTensor/Test_KerrSphericalHarmonic.cpp [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/AnalyticData/ScalarTensor/KerrSphericalHarmonic.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+#include "Parallel/Printf.hpp"
```
Comment (nilsdeppe) [inline tests/Unit/PointwiseFunctions/AnalyticData/ScalarTensor/Test_KerrSphericalHarmonic.cpp]: remove

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM! A few minor suggestions. You can squash any fixes in immediately :)

**Comment by guilara** [comment]
@nilsdeppe thanks for the review. I've implemented all the suggestions and improved a bit the doc

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**